src/transformers/models/sam/modeling_sam.py

```diff
@@ -634,8 +638,20 @@ def _embed_points(self, points: torch.Tensor, labels: torch.Tensor, pad: bool) -
             torch.tensor(0.0, dtype=point_embedding.dtype, device=point_embedding.device),
         )
 
-        point_embedding[labels == 0] += self.point_embed[0].weight
-        point_embedding[labels == 1] += self.point_embed[1].weight
+        point_embedding = torch.where(
+                (labels == 0)[:, :, :, None],
+                point_embedding + self.point_embed[0].weight[None, None, :, :],
+                point_embedding
+        )
+
+        point_embedding = torch.where(
+                (labels == 1)[:, :, :, None],
+                point_embedding + self.point_embed[1].weight[None, None, :, :],
+                point_embedding
+        )
```

Comment:
This is the actual fix for the ONNX export to work in `optimum`. 

The rest of the changes do not prevent the ONNX export per-se, but users can face issues (the export pass, but it does not mean that it will work in every case).